### PR TITLE
Cloning Options and so they don't get modified when a new instance is created.

### LINKED ISSUE #4136: [Event Hubs] Clone retry policy and other builder parameters
* If any of the parameters are mutable, we should create a clone of them before creating the client/options/etc.

### PR DESCRIPTION
**Pull request by conniey**
* Clone EventHubProducerOptions, Builder options, Retry, so if any modifications occur to them, it does not affect the new instance.
* Fix spelling error in `MessageConstants.ABSOLUTE_EXPRITY_TIME`
* Cloneable classes implement `Cloneable` and expose `public Object clone() { }`

Fixes #4136, #3714

### CODE REVIEW

File: core/azure-core-amqp/pom.xml [hunk 1]
```diff
@@ -39,9 +39,9 @@
     <url>https://github.com/Azure/azure-sdk-for-java</url>
     <connection>scm:git:https://github.com/Azure/azure-sdk-for-java.git</connection>
     <developerConnection>scm:git:https://github.com/Azure/azure-sdk-for-java.git</developerConnection>
-    <tag>v1.0.0-preview.1</tag>
+    <tag>v1.0.0-preview.3</tag>
```
Comment (weshaggard) [inline core/azure-core-amqp/pom.xml]: We should remove the tag as it doesn't add anything to duplicate the version here.

File: core/azure-core-amqp/pom.xml [hunk 2]
```diff
@@ -7,13 +7,13 @@
   <parent>
     <groupId>com.azure</groupId>
     <artifactId>azure-core-parent</artifactId>
-    <version>1.0.0-preview.1</version>
+    <version>1.1.0-SNAPSHOT</version>
```
Comment (weshaggard) [inline core/azure-core-amqp/pom.xml]: For now this should be 1.1.0 to match what is in the repo, at least until there are changes and it's version is bumped.

File: core/azure-core-amqp/pom.xml [hunk 3]
```diff
@@ -7,13 +7,13 @@
...
     <relativePath>../pom.xml</relativePath>
   </parent>
 
   <groupId>com.azure</groupId>
   <artifactId>azure-core-amqp</artifactId>
-  <version>1.0.0-preview.1</version>
```
Comment (weshaggard) [inline core/azure-core-amqp/pom.xml]: Should be changed to `1.0.0-preview.3`. The diff here looks a little odd as it is `1.0.0-preview.2` in master (https://github.com/Azure/azure-sdk-for-java/blob/master/core/azure-core-amqp/pom.xml#L16), you might want to rebase your changes on the latest master as I think some of these were bumped as part of a respin for the first preview.

File: core/azure-core-amqp/pom.xml [hunk 4]
```diff
@@ -58,7 +58,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core</artifactId>
-      <version>1.0.0-preview.1</version>
+      <version>1.1.0-SNAPSHOT</version>
```
Comment (weshaggard) [inline core/azure-core-amqp/pom.xml]: Should be `1.0.0-preview.3` and we should update azure-core version to match.

File: core/azure-core-amqp/src/main/java/com/azure/core/amqp/Retry.java [hunk 5]
```diff
@@ -132,6 +132,11 @@ public Duration getNextRetryInterval(Exception lastException, Duration remaining
         return this.calculateNextRetryInterval(lastException, remainingTime, baseWaitTime, this.getRetryCount());
     }
 
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        return super.clone();
```
Comment (JonathanGiles) [inline core/azure-core-amqp/src/main/java/com/azure/core/amqp/Retry.java]: Have you tested that this works as expected?

Comment (JonathanGiles) [reply core/azure-core-amqp/src/main/java/com/azure/core/amqp/Retry.java]: It looks like the answer is yes as I scroll further down...

File: eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConsumerOptions.java [hunk 6]
```diff
@@ -211,10 +214,18 @@ public EventHubConsumerOptions clone() {
             clone = new EventHubConsumerOptions();
         }
 
+        if (retry != null) {
+            try {
+                clone.retry((Retry) retry.clone());
+            } catch (CloneNotSupportedException e) {
+                logger.asWarning().log("Unable to create clone of retry.", e);
```
Comment (JonathanGiles) [inline eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConsumerOptions.java]: warning or error?

Comment (conniey) [reply eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConsumerOptions.java]: Changed to error.

**Pull request merged**